src/app/config/projectProperties.json

```diff
@@ -1,68 +1,82 @@
 {
     "projectTypes": {
         "angular": {
-            "name": "angular",
             "displayname": "Office Add-in project using Angular framework",
-            "repolocation": "",
+            "templates": {
+                "typescript": {
+                    "repository": ""},
+                "javascript": {
+                    "repository": ""}
+                },
             "typescript": true,
             "javascript": true
         },
         "excel-functions": {
-            "name": "excel-functions",
             "displayname": "Excel Custom Functions Add-in project (Preview: Requires the Insider channel for Excel)",
-            "repolocation": "",
-            "typescript": true,
+            "templates": {
+                "typescript": {
```

Comment:
You could simplify this by using whether a "typescript" or "javascript" template is available, and not need the bool property. This would mean to have only javascript template right now for excel-functions project type, and not have either for manifest project type. The others would have both.